fix: fix progress window for 'minikube start'

ComradeProgrammer opened the pull request:
FIX #73 
feat: set status to starting when starting minikube

When minikube is being started (by minikube-gui), both the system tray and the main window will show status "Starting"

Like this:
![06201](https://github.com/kubernetes-sigs/minikube-gui/assets/46831212/fdcc7d86-67f5-4a21-a2ae-06255e4646f4)

![06202](https://github.com/kubernetes-sigs/minikube-gui/assets/46831212/4ac379c0-e571-428f-a8e6-2d9cedf0d5cd)

medyagh reviewed (changes requested):
@ComradeProgrammer thank you for this PR, 

can we make it that, if "minikube start" fails, will it revert back to what the status was before?

to test it you can start minikube with Gui, and then Kill the minikube in terminal, and u should expect some failure...and it should no longer say starting..

to test it

ComradeProgrammer commented:
I am indeed always ambitious, but this feature is definitely not a good place to be ambitious(laugh, it is a joke).

After reading the code I just found that Steven has previously implemented a more complicated progress tracking window like this:

![Screenshot from 2023-06-25 22-39-37](https://github.com/kubernetes-sigs/minikube-gui/assets/46831212/809bfe9c-6db3-414a-abb1-76e39516c50b)

When the minikube cluster is starting, real time outputs generated by `-o json` are sent to this window by `CommandRunner::output` signal, which is connected with `Operator::commandOutput`. After that, `Operator::commandOutput` updates the text and progress bar of this window.

However, this window, which had already implemented the required feature, was not visible at all before. It is because that when this window was created , the WindowFlags was incorrectly set.  It should be `Qt::Dialog| Qt::FramelessWindowHint` (which equals to `m_dialog->windowFlags() | Qt::FramelessWindowHint` in this PR), not solely `Qt::FramelessWindowHint`, which make this window disappear.

I just amended my PR and fixed this bug.

By the way, I am always looking forward to contributing to some more complicated features, if possible.  :)

medyagh commented:
> I am indeed always ambitious, but this feature is definitely not a good place to be ambitious(laugh, it is a joke).
> 
> After reading the code I just found that Steven has previously implemented a more complicated progress tracking window like this:
> 
> ![Screenshot from 2023-06-25 22-39-37](https://user-images.githubusercontent.com/46831212/248565056-809bfe9c-6db3-414a-abb1-76e39516c50b.png)
> 
> When the minikube cluster is starting, real time outputs generated by `-o json` are sent to this window by `CommandRunner::output` signal, which is connected with `Operator::commandOutput`. After that, `Operator::commandOutput` updates the text and progress bar of this window.
> 
> However, this window, which had already implemented the required feature, was not visible at all before. It was because that when this window was created , the WindowFlags was incorrectly set. It should be `Qt::Dialog| Qt::FramelessWindowHint` (which equals to `m_dialog->windowFlags() | Qt::FramelessWindowHint` in this PR), not solely `Qt::FramelessWindowHint`, which make this window disappear.
> 
> I just amended my PR and fixed this bug.
> 
> By the way, I am always looking forward to contributing to some more complicated features, if possible. :)
> 
> @medyagh @spowelljr

Excellent ! @ComradeProgrammer good work here ! I appreciate your contribution and look forward to find good issues for you to work on

spowelljr commented on src/operator.cpp:
```diff
@@ -151,7 +151,6 @@ void Operator::createCluster(QStringList args)
 
 void Operator::startCommandStarting()
 {
-    commandStarting();
```
This doesn't fix it for me, for me, on main or this PR I get the same result. The start window shows up on the first start, but any subsequent starts the loading window doesn't show up.

spowelljr replied:
Fixed with https://github.com/kubernetes-sigs/minikube-gui/pull/86, could you revert removing this line and then it'll be good to merge

spowelljr commented:
Thanks for fixing the progress loading bar!

However, for the status showing `Starting` I don't quite understand the change.

You're not manually setting the text to `Starting` and `minikube profile list -o json` doesn't get called until after the start command has completed, so how is the text becoming `Starting`?

ComradeProgrammer commented:
> Thanks for fixing the progress loading bar!
> 
> However, for the status showing `Starting` I don't quite understand the change.
> 
> You're not manually setting the text to `Starting` and `minikube profile list -o json` doesn't get called until after the start command has completed, so how is the text becoming `Starting`?

Oh sorry I removed that part of code.

On the one hand, perhaps with this progress bar blocking the main window, we no longer need to change this text to make user understand minikube is being started.

On the other hand, because "Running" is not a status returned from minikube, so changing this text is quite complicated (we have to consider how to change the text back if this `minikube start ` command fails)

I have changed the picture, which no longer shows a "Running status"
